apache/nifi
# NIFI-6196 Upgrade version of Jetty

This PR should be merged after https://github.com/apache/nifi/pull/3416 to avoid merge conflicts due to regenerated certs in nifi-standard-processor.

While reviewing code, it is important to verify that server certs are not reused as client certs.  This PR updates several tests that reused server certs, but more investigation is needed for other areas of code that reuse them.

To test this PR:
- Specific components that use Jetty:  ListenHTTP, HandleHttpRequest, InvokeHTTP, JettyWebSocketClient, JettyWebSocketServer, TlsCertificateAuthorityService

Each component should be tested in the following scenarios:
- Verify an unsecured standalone instance of NiFi
- Verify a secured anonymous standalone instance of NiFi
- Verify a secured standalone instance of NiFi
- Verify a secured cluster of NiFi

To test the TlsCertificateAuthorityService, certs for secured NiFi instances should be generated by a running NiFi CA service.

Please see https://issues.apache.org/jira/browse/NIFI-6196 for the list of issues encountered and their resolutions implemented in this PR.

#### **NOTE:** Please retain the flows used to verify this PR.  They can be reused for verifying functionality when NiFi supports builds on Java 11.
----

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.



### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [x] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
